Lib/typing.py

```diff
@@ -785,8 +789,28 @@ def __repr__(self):
             module_repr = f', module={self.__forward_module__!r}'
         return f'ForwardRef({self.__forward_arg__!r}{module_repr})'
 
-class _TypeVarLike:
-    """Mixin for TypeVar-like types (TypeVar and ParamSpec)."""
+
+def _is_unpacked_typevartuple(x):
+    return (
+            isinstance(x, _UnpackGenericAlias)
+            # If x is Unpack[tuple[...]], __parameters__ will be empty.
+            and x.__parameters__
+            and isinstance(x.__parameters__[0], TypeVarTuple)
+    )
+
+
+def _is_typevar_like(x):
+    return _is_unpacked_typevartuple(x) or isinstance(x, (TypeVar, ParamSpec))
```

Comment:
```suggestion
    return isinstance(x, (TypeVar, ParamSpec)) or _is_unpacked_typevartuple(x)
```
Micro-optimization (probably most calls will be with TypeVars, so do that check first).

Reply:
Done.